core/src/mindustry/world/blocks/defense/ShockwaveTower.java

```diff
@@ -0,0 +1,111 @@
+package mindustry.world.blocks.defense;
+
+import arc.math.*;
+import arc.util.*;
+import arc.struct.*;
+import arc.graphics.*;
+import arc.graphics.g2d.*;
+import arc.audio.*;
+import mindustry.gen.*;
+import mindustry.graphics.*;
+import mindustry.world.meta.*;
+import mindustry.world.*;
+import mindustry.entities.*;
+import mindustry.annotations.Annotations.Load;
+
+import static arc.graphics.g2d.Draw.color;
+import static arc.graphics.g2d.Lines.*;
+import static arc.math.Angles.randLenVectors;
+import static mindustry.Vars.tilesize;
+
+public class ShockwaveTower extends Block {

```

Comment:
Formatting in this entire file is off, see the [style guidelines.](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md#style-guidelines)